presto-main/src/main/java/com/facebook/presto/operator/NestedLoopJoinOperator.java

```diff
@@ -193,18 +201,37 @@ public Page getOutput()
         }
 
         if (nestedLoopPageBuilder != null && nestedLoopPageBuilder.hasNext()) {
-            return nestedLoopPageBuilder.next();
+            return applyFilter(nestedLoopPageBuilder.next());
         }
 
         if (buildPageIterator.hasNext()) {
             nestedLoopPageBuilder = new NestedLoopPageBuilder(probePage, buildPageIterator.next());
-            return nestedLoopPageBuilder.next();
+            return applyFilter(nestedLoopPageBuilder.next());
         }
 
         probePage = null;
         return null;
     }
 
+    private Page applyFilter(Page page)
+    {
+        if (!pageFilter.isPresent()) {
+            return page;
+        }
+
+        SelectedPositions selectedPositions = pageFilter.get().filter(operatorContext.getSession().toConnectorSession(), pageFilter.get().getInputChannels().getInputChannels(page));
+        if (selectedPositions.isList()) {
+            int[] retainedPositions = Arrays.copyOfRange(
+                    selectedPositions.getPositions(),
+                    selectedPositions.getOffset(),
+                    selectedPositions.getOffset() + selectedPositions.size());
+            return page.getPositions(retainedPositions);
```

Comment:
After https://github.com/prestodb/presto/pull/9594 lands, please update to the new api